### DISCUSSION TITLE
str, DynamicArray 是否考虑支持负索引( Negative Indexing)？

### DISCUSSION THREAD

**Discussion by microctar** [prajna-lang/prajna#28] [Ideas]
当前DynamicArray实现中，capacity, length的类型为i64，
当前str实现中，size类型为i64。

那么parajna将来是否考虑支持负索引？

**Comment by microctar** [comment]
如不打算支持负索引，应当将类型设置为u64

**Reply by zhangzhimin** [reply]
不支持负索引, 会影响性能的.
没必要设置为u64, i64不会越界, u64不小心减了就越界了

**Reply by microctar** [reply]
> 不支持负索引, 会影响性能的. 没必要设置为u64, i64不会越界, u64不小心减了就越界了

那么i64也存在同样问题，也会不小心出现负数, 同样越界. 
这是C的老毛病了.

**Reply by zhangzhimin** [reply]
size_t idx = 0;
auto v = idx -1; v直接就越界了.
int64_t是不会的.

要考虑运算的完备性, uint运算其实是不完备的, 使用场景是需要受到控制的
